fvm/Equation/DiffusionTerm.set.cpp

```diff
@@ -44,12 +44,15 @@
                 /////////////////////////
                 // RADIUS
                 /////////////////////////
-
                 // Trapping BC: even if the cell is not ignorable, it may still 
                 // be such that the radial flux should be mirrored 
-                if(!isNegativeTrappedRadial && ( Drr(ir, i, j, drr) || Drr(ir+1, i, j, drr) ) ){
+                
+                // For the Ions in DREAM the transient term is defined such that the 
+                // diffusion coefficients should must be negative to get the correct
+                // sign on the diffusion term, which is why we use abs(Drr) here. 
+                // This should however probably be reworked in a better way...
+                if(!isNegativeTrappedRadial && ( abs(Drr(ir, i, j, drr)) || abs(Drr(ir+1, i, j, drr)) ) ){
```

Comment:
May want to explicitly give ``std::abs()``. The compiler may otherwise silently replace the ``std::abs()`` function with the old ``abs()`` function from the C standard library, which only operates on integers.

I would also favour explicitly writing ``std::abs(...) > 0`` to emphasize what is being tested here.